plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -428,20 +413,39 @@ export class GitLabClient {
       }
     }
 
-    const response = await fetch(
-      request.toString(),
-      getGitLabRequestOptions(this.config),
-    );
-
-    if (!response.ok) {
-      throw new Error(
-        `Unexpected response when fetching ${request.toString()}. Expected 200 but got ${
-          response.status
-        } - ${response.statusText}`,
+    for (let i = 0; ; i++) {
+      this.logger.debug(`Fetching: ${request.toString()}`);
+      const response = await fetch(
+        request.toString(),
+        getGitLabRequestOptions(this.config),
       );
-    }
 
-    return response.json();
+      if (response.status === tooManyRequests && i < maxRetry) {
+        const retryAfter = response.headers.get('Retry-After');
+
+        if (retryAfter) {
```

Comment:
Is it only going to retry after value is returned in the response, just to be clear?

Reply:
@camilaibs My thought process was:
* `429` status received (`const tooManyRequests`)
* Look for `Retry-After` response header
* If said header's value is truthy, sleep for the specified duration

So essentially if there is no `Retry-After` header the retry will occur immediately. We could replace this with some default sleep duration. Does that answer your question?